src/twisted/conch/ssh/keys.py

```diff
@@ -1241,8 +1402,20 @@ def verify(self, signature, data):
                 k = k.public_key()
             verifier = k.verifier(
                 signature, hashes.SHA1())
-        else:
-            raise BadKeyError("unknown key type %s" % (self.type(),))
+
+        elif keyType == 'EC':
+            k = self._keyObject
+            if not self.isPublic():
+                k=k.public_key()
+            keySize = self.size()
+            if keySize <= 256:
+                hashSize = hashes.SHA256()
+            elif keySize <= 384:
+                hashSize = hashes.SHA384()
+            else:
+                hashSize = hashes.SHA512()
```

Comment:
I'm a little perplexed by this block of code. Care to put a comment above it explaining why we choose the hash size this way?